test/receiveAndDeleteMode.spec.ts

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import "mocha";
+import * as chai from "chai";
+const should = chai.should();
+import * as chaiAsPromised from "chai-as-promised";
+import * as dotenv from "dotenv";
+dotenv.config();
+chai.use(chaiAsPromised);
+import {
+  Namespace,
+  QueueClient,
+  SendableMessageInfo,
+  generateUuid,
+  TopicClient,
+  SubscriptionClient,
+  delay,
+  ServiceBusMessage,
+  ReceiveMode
+} from "../lib";
+
+const testMessages: SendableMessageInfo[] = [
+  {
+    body: "hello1",
+    messageId: `test message ${generateUuid()}`
+  },
+  {
+    body: "hello2",
+    messageId: `test message ${generateUuid()}`
+  }
+];
+
+async function testPeekMsgsLength(
+  client: QueueClient | SubscriptionClient,
+  expectedPeekLength: number
+): Promise<void> {
+  const peekedMsgs = await client.peek(expectedPeekLength + 1);
+  should.equal(peekedMsgs.length, expectedPeekLength);
```

Comment:
Please add a message here like we do in https://github.com/Azure/azure-service-bus-node/blob/master/test/batchReceiver.spec.ts#L41
This will help us in figuring out which check failed